src/ring_theory/algebraic.lean

```diff
@@ -0,0 +1,89 @@
+/-
+Copyright (c) 2019 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import ring_theory.integral_closure
+
+/-!
+# Algebraic elements and algebraic extensions
+
+An element of an R-algebra is algebraic over R if it is the root of a nonzero polynomial.
+An R-algebra is algebraic over R if and only if all its elements are algebraic over R.
+The main result in this file proves transitivity of algebraicity:
+a tower of algebraic field extensions is algebraic.
+-/
+
+universe variables u v
+
+open_locale classical
+open polynomial
+
+section
+variables (R : Type u) {A : Type v} [comm_ring R] [comm_ring A] [algebra R A]
+
+/-- An element of an R-algebra is algebraic over R if it is the root of a nonzero polynomial. -/
+def is_algebraic (x : A) : Prop :=
+∃ p : polynomial R, p ≠ 0 ∧ aeval R A x p = 0
+
+variables {R}
+
+/-- A subalgebra is algebraic if all its elements are algebraic. -/
+def subalgebra.is_algebraic (S : subalgebra R A) : Prop := ∀ x ∈ S, is_algebraic R x
```

Comment:
Is there a reason to introduce `is_algebraic` for subalgebras first? It looks like "compact subset" again.
Why not just define `algebra.is_algebraic` directly and then say a subalgebra is algebraic if it is algebraic as an algebra?

Reply:
Either way I think you will eventually want to prove that a subalgebra `S` is algebraic in the sense of `algebra.is_algebraic` if and only if it is algebraic in the sense of this current `subalgebra.is_algebraic` (right?). To me `algebra.is_algebraic` seems like the primary notion so defining it directly looks better.

Reply:
More or less done. They now all have elementwise definitions and `iff`s relating the various definitions.